pdns/pdns_recursor.cc

```diff
@@ -3809,19 +3816,18 @@ template<class T> T broadcastAccFunction(const boost::function<T*()>& func)
     tmsg->func = [func]{ return voider<T>(func); };
     tmsg->wantAnswer = true;
 
-    if(write(tps.writeToThread, &tmsg, sizeof(tmsg)) != sizeof(tmsg)) {
+    if (write(tps.writeToThread, &tmsg, sizeof(tmsg)) != sizeof(tmsg)) {
       delete tmsg;
       unixDie("write to thread pipe returned wrong size or error");
     }
 
-    T* resp = nullptr;
-    if(read(tps.readFromThread, &resp, sizeof(resp)) != sizeof(resp))
+    if (read(tps.readFromThread, &resp, sizeof(resp)) != sizeof(resp)) {
+      delete tmsg;
```

Comment:
I'm not so sure it's a good idea to delete it now. It has successfully been written to the pipe, the fact that we are not getting an answer might not mean that the other end has not received it? Perhaps it did, got an exception but already deleted the object? A small leak might be better than a double-free?

Reply:
Right, the responsibility is at the other end.